### ISSUE TITLE
TIN on points with short distance between them returns empty

### ISSUE THREAD

**Issue by shayb-datumate** [Turfjs/turf#1809]
Hi guys,

I'm trying to calculate the surface area of a given polygon with high accuracy (2 cm).
The way I'm doing it is by generating point samples (using the `pointGrid` method with 0.02 meter `cellSide`), then generate a TIN on these points (using the `tin` method) and calculate each triangle's 3D area.
Unfortunately, `tin` always returns an empty feature collection, unless I increase the `cellSide`.

Here's a fiddle demonstrating this:
https://jsfiddle.net/wjo3pbv1

Am I doing something wrong here? Is there a better way to achieve this?

**Comment by stebogit** [comment]
@shayb-datumate Turf tries to follow the GeoJSON specs, which [suggests](https://tools.ietf.org/html/rfc7946#section-11.2) a coordinates precision of 6 decimal places corresponding to about 10cm precision.
For this reason some modules might round or truncate the decimals in the coordinates, most of the times to avoid tests to fail due to precision errors.
Not sure what's going on with the `tin` module, but in general with Turf you could encounter other issues related to precision in your particular case.

**Comment by rowanwins** [comment]
Thanks for the report @shayb-datumate - is there something wrong with using the [area](https://turfjs.org/docs/#area) module?

**Comment by shayb-datumate** [comment]
@rowanwins As far as I understand, `area` calculates the 2D area of polygons, not 3D, which is what I need. Am I wrong?

**Comment by stebogit** [comment]
@shayb-datumate that is correct.
What you are looking for would be then the volume of a 3D object, which you can only represent as a (polygonal) projection on a map. However, if the height of your object is constant you can simply multiply the polygon's area by the height to get its volume.

**Comment by shayb-datumate** [comment]
I'm not trying to calculate the volume of a 3D object, but the surface area of a polygon, i.e. the area of a polygon that is clamped to a terrain, where each point in it have different height.

Here's an image I found online that sort of explains what I mean:
![image](https://user-images.githubusercontent.com/52623423/71656721-2d0c5e00-2d45-11ea-9df9-fb74d4b15d52.png)

**Comment by rowanwins** [comment]
Mmm thats not doable within turf at the moment

**Comment by shayb-datumate** [comment]
Yes I know, that's why I'm using `tin` and calculate the area of each resulting triangle's 3D area myself.
Is it possible to fix the above issue with `tin` which returns no triangles when supplying it with very close distance between them? 🙏🏻

**Comment by stebogit** [comment]
@shayb-datumate have you tried with `@turf/isobannds`?
